ethereum/core/src/test/java/tech/pegasys/pantheon/ethereum/storage/keyvalue/RocksDbStorageProviderTest.java

```diff
@@ -0,0 +1,69 @@
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.ethereum.storage.keyvalue;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.when;
+
+import tech.pegasys.pantheon.metrics.MetricsSystem;
+import tech.pegasys.pantheon.metrics.noop.NoOpMetricsSystem;
+import tech.pegasys.pantheon.services.kvstore.RocksDbConfiguration;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class RocksDbStorageProviderTest {
+
+  @Mock private RocksDbConfiguration rocksDbConfiguration;
+  @Rule public final TemporaryFolder temporaryFolder = new TemporaryFolder();
+  private final MetricsSystem metricsSystem = new NoOpMetricsSystem();
+
+  @Test
+  public void shouldCreateCorrectMetadataFileForLatestVersion() throws Exception {
+    final Path tempDatabaseDir = temporaryFolder.newFolder().toPath().resolve("db");
+    when(rocksDbConfiguration.getDatabaseDir()).thenReturn(tempDatabaseDir);
+    RocksDbStorageProvider.create(rocksDbConfiguration, metricsSystem);
+    assertEquals(
+        RocksDbStorageProvider.CURRENT_VERSION,
+        DatabaseMetadata.fromDirectory(rocksDbConfiguration.getDatabaseDir()).getVersion());
+  }
+
+  @Test
+  public void shouldDetectVersion0DatabaseIfNoMetadataFileFound() throws Exception {
+    final Path tempDatabaseDir = temporaryFolder.newFolder().toPath().resolve("db");
+    Files.createDirectories(tempDatabaseDir);
+    tempDatabaseDir.resolve("IDENTITY").toFile().createNewFile();
+    when(rocksDbConfiguration.getDatabaseDir()).thenReturn(tempDatabaseDir);
+    RocksDbStorageProvider.create(rocksDbConfiguration, metricsSystem);
+    assertEquals(0, DatabaseMetadata.fromDirectory(tempDatabaseDir).getVersion());
+  }
+
+  @Test
+  public void shouldDetectCorrectVersionIfMetadataFileExists() throws Exception {
+    final Path tempDatabaseDir = temporaryFolder.newFolder().toPath().resolve("db");
+    Files.createDirectories(tempDatabaseDir);
+    tempDatabaseDir.resolve("IDENTITY").toFile().createNewFile();
+    new DatabaseMetadata(1).writeToDirectory(tempDatabaseDir);
+    when(rocksDbConfiguration.getDatabaseDir()).thenReturn(tempDatabaseDir);
+    RocksDbStorageProvider.create(rocksDbConfiguration, metricsSystem);
+    assertEquals(1, DatabaseMetadata.fromDirectory(tempDatabaseDir).getVersion());
+  }
```

Comment:
per my previous comment, we need a case where the file exists and is invalid.  (corrupt, has unsupported version number, etc.)